modules/core/test/test_hal_core.cpp

```diff
@@ -42,168 +42,187 @@
+        memcpy(x.ptr(), b.ptr(), bsize);
+        memcpy(a.ptr(), a0.ptr(), asize);
+
+        double t = (double)getTickCount();
+        switch (nfunc)
         {
-            memcpy(x.ptr(), b.ptr(), bsize);
-            memcpy(a.ptr(), a0.ptr(), asize);
-
-            double t = (double)getTickCount();
-            switch (nfunc)
-            {
-            case HAL_LU:
-                if( depth == CV_32F )
-                    hal::LU32f(a.ptr<float>(), a.step, size, x.ptr<float>(), x.step, 1);
-                else
-                    hal::LU64f(a.ptr<double>(), a.step, size, x.ptr<double>(), x.step, 1);
-                break;
-            case HAL_CHOL:
-                if( depth == CV_32F )
-                    hal::Cholesky32f(a.ptr<float>(), a.step, size, x.ptr<float>(), x.step, 1);
-                else
-                    hal::Cholesky64f(a.ptr<double>(), a.step, size, x.ptr<double>(), x.step, 1);
-                break;
-            default:
-                CV_Error(Error::StsBadArg, "unknown function");
-            }
-            t = (double)getTickCount() - t;
-            min_hal_t = std::min(min_hal_t, t);
-
-            t = (double)getTickCount();
-            bool solveStatus = solve(a0, b, x0, (nfunc == HAL_LU ? DECOMP_LU : DECOMP_CHOLESKY));
-            t = (double)getTickCount() - t;
-            EXPECT_TRUE(solveStatus);
-            min_ocv_t = std::min(min_ocv_t, t);
+        case HAL_LU:
+            if( depth == CV_32F )
+                hal::LU32f(a.ptr<float>(), a.step, size, x.ptr<float>(), x.step, 1);
+            else
+                hal::LU64f(a.ptr<double>(), a.step, size, x.ptr<double>(), x.step, 1);
+            break;
+        case HAL_CHOL:
+            if( depth == CV_32F )
+                hal::Cholesky32f(a.ptr<float>(), a.step, size, x.ptr<float>(), x.step, 1);
+            else
+                hal::Cholesky64f(a.ptr<double>(), a.step, size, x.ptr<double>(), x.step, 1);
+            break;
+        default:
+            CV_Error(Error::StsBadArg, "unknown function");
         }
-        //std::cout << "x: " << Mat(x.t()) << std::endl;
-        //std::cout << "x0: " << Mat(x0.t()) << std::endl;
-
-        EXPECT_LE(cvtest::norm(x, x0, NORM_INF | NORM_RELATIVE), eps)
-            << "x:  " << Mat(x.t())
-            << "\nx0: " << Mat(x0.t())
-            << "\na0: " << a0
-            << "\nb: " << b;
-
-        double freq = getTickFrequency();
-        printf("%s (%d x %d, %s): hal time=%.2fusec, ocv time=%.2fusec\n",
-               (nfunc == HAL_LU ? "LU" : nfunc == HAL_CHOL ? "Cholesky" : "???"),
-               size, size,
-               (depth == CV_32F ? "f32" : "f64"),
-               min_hal_t*1e6/freq, min_ocv_t*1e6/freq);
+        t = (double)getTickCount() - t;
+        min_hal_t = std::min(min_hal_t, t);
+
+        t = (double)getTickCount();
+        bool solveStatus = solve(a0, b, x0, (nfunc == HAL_LU ? DECOMP_LU : DECOMP_CHOLESKY));
```

Comment:
This test is weak because it tests explicit HAL call and the same HAL call at https://github.com/opencv/opencv/blob/8e65075c1e6810c3ffbf38093f4e5fc167d76ea7/modules/core/src/matrix_decomp.cpp#L187

Reply:
I've inverted the test but keep comparison between `A * x`  and `A * x0` because test `x` vs `x0` gives bigger error on `size=15`:
```
[ RUN      ] Core_HAL/mat_decomp.accuracy/7, where GetParam() = (5, Cholesky, 15)
/home/d.kurtaev/opencv/modules/core/test/test_hal_core.cpp:189: Failure
Expected: (cvtest::norm(x, x0, NORM_INF | NORM_RELATIVE)) <= (eps), actual: 0.03707 vs 1e-05
```